diff -Naur gcc/gcc/config/i386/t-x86_64-elf gcc_bareflank/gcc/config/i386/t-x86_64-elf
--- gcc/gcc/config/i386/t-x86_64-elf	1969-12-31 16:00:00.000000000 -0800
+++ gcc_bareflank/gcc/config/i386/t-x86_64-elf	2016-01-20 04:31:05.551783604 -0800
@@ -0,0 +1,2 @@
+MULTILIB_OPTIONS += mno-red-zone
+MULTILIB_DIRNAMES += no-red-zone
diff -Naur gcc/gcc/config.gcc gcc_bareflank/gcc/config.gcc
--- gcc/gcc/config.gcc	2016-01-20 04:30:07.866086228 -0800
+++ gcc_bareflank/gcc/config.gcc	2016-01-20 04:32:17.992897101 -0800
@@ -1369,6 +1369,7 @@
 	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h newlib-stdint.h i386/i386elf.h"
 	;;
 x86_64-*-elf*)
+	tmake_file="${tmake_file} i386/t-x86_64-elf"
 	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h newlib-stdint.h i386/i386elf.h i386/x86-64.h"
 	;;
 i[34567]86-*-rdos*)
