diff -Naur source_libcxx_original/CMakeLists.txt source_libcxx/CMakeLists.txt
--- source_libcxx_original/CMakeLists.txt	2016-09-06 11:50:25.003974957 -0600
+++ source_libcxx/CMakeLists.txt	2016-09-06 12:23:30.835600782 -0600
@@ -310,9 +310,8 @@
 add_definitions(-D_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER)
 add_compile_flags_if_supported(
     -Wall -W -Wwrite-strings
-    -Wno-unused-parameter -Wno-long-long -Wno-user-defined-literals
-    -Wno-covered-switch-default
-    -Werror=return-type)
+    -Wno-unused-parameter -Wno-long-long -Wno-strict-aliasing -Wno-attributes
+    -Werror=return-type -Wno-c++14-compat)
 if (LIBCXX_ENABLE_WERROR)
   add_compile_flags_if_supported(-Werror)
   add_compile_flags_if_supported(-WX)
diff -Naur source_libcxx_original/include/atomic source_libcxx/include/atomic
--- source_libcxx_original/include/atomic	2016-09-06 11:50:25.003974957 -0600
+++ source_libcxx/include/atomic	2016-09-06 12:23:30.727606751 -0600
@@ -589,6 +589,9 @@
     : __a_value(value) {}
   _Tp __a_value;
 };
+#ifdef _Atomic
+#undef _Atomic
+#endif
 #define _Atomic(x) __gcc_atomic::__gcc_atomic_t<x>
 
 template <typename _Tp> _Tp __create();
diff -Naur source_libcxx_original/src/strstream.cpp source_libcxx/src/strstream.cpp
--- source_libcxx_original/src/strstream.cpp	2016-09-06 11:50:25.007974684 -0600
+++ source_libcxx/src/strstream.cpp	2016-09-06 12:23:30.727606751 -0600
@@ -253,7 +253,7 @@
         legal = false;
     if (legal)
     {
-        off_type newoff;
+        off_type newoff = 0;
         char* seekhigh = epptr() ? epptr() : egptr();
         switch (__way)
         {
