diff -Naur source_libcxxabi_original/CMakeLists.txt source_libcxxabi/CMakeLists.txt
--- source_libcxxabi_original/CMakeLists.txt	2016-09-06 08:38:50.645647621 -0600
+++ source_libcxxabi/CMakeLists.txt	2016-09-06 10:18:05.257087827 -0600
@@ -262,8 +262,8 @@
 append_if(LIBCXXABI_COMPILE_FLAGS LIBCXXABI_HAS_WSHORTEN_64_TO_32_FLAG -Wshorten-64-to-32)
 append_if(LIBCXXABI_COMPILE_FLAGS LIBCXXABI_HAS_WSIGN_COMPARE_FLAG -Wsign-compare)
 append_if(LIBCXXABI_COMPILE_FLAGS LIBCXXABI_HAS_WSIGN_CONVERSION_FLAG -Wsign-conversion)
-append_if(LIBCXXABI_COMPILE_FLAGS LIBCXXABI_HAS_WSTRICT_ALIASING_FLAG -Wstrict-aliasing=2)
-append_if(LIBCXXABI_COMPILE_FLAGS LIBCXXABI_HAS_WSTRICT_OVERFLOW_FLAG -Wstrict-overflow=4)
+#append_if(LIBCXXABI_COMPILE_FLAGS LIBCXXABI_HAS_WSTRICT_ALIASING_FLAG -Wstrict-aliasing=2)
+#append_if(LIBCXXABI_COMPILE_FLAGS LIBCXXABI_HAS_WSTRICT_OVERFLOW_FLAG -Wstrict-overflow=4)
 append_if(LIBCXXABI_COMPILE_FLAGS LIBCXXABI_HAS_WUNUSED_PARAMETER_FLAG -Wunused-parameter)
 append_if(LIBCXXABI_COMPILE_FLAGS LIBCXXABI_HAS_WUNUSED_VARIABLE_FLAG -Wunused-variable)
 append_if(LIBCXXABI_COMPILE_FLAGS LIBCXXABI_HAS_WWRITE_STRINGS_FLAG -Wwrite-strings)
diff -Naur source_libcxxabi_original/src/cxa_handlers.cpp source_libcxxabi/src/cxa_handlers.cpp
--- source_libcxxabi_original/src/cxa_handlers.cpp	2016-09-06 08:38:50.645647621 -0600
+++ source_libcxxabi/src/cxa_handlers.cpp	2016-09-06 10:18:05.253088009 -0600
@@ -104,7 +104,7 @@
 
 // In the future this will become:
 // std::atomic<std::new_handler>  __cxa_new_handler(0);
-extern "C" _LIBCXXABI_DATA_VIS new_handler __cxa_new_handler = 0;
+extern "C" { _LIBCXXABI_DATA_VIS new_handler __cxa_new_handler = 0; }
 
 new_handler
 set_new_handler(new_handler handler) _NOEXCEPT
diff -Naur source_libcxxabi_original/src/private_typeinfo.cpp source_libcxxabi/src/private_typeinfo.cpp
--- source_libcxxabi_original/src/private_typeinfo.cpp	2016-09-06 08:38:50.645647621 -0600
+++ source_libcxxabi/src/private_typeinfo.cpp	2016-09-06 10:18:05.253088009 -0600
@@ -37,6 +37,10 @@
 
 #include <string.h>
 
+#ifdef __GNUC__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+#endif
 
 #ifdef _LIBCXX_DYNAMIC_FALLBACK
 #include "abort_message.h"
@@ -1277,3 +1281,9 @@
 #pragma GCC visibility pop
 
 }  // __cxxabiv1
+
+
+#ifdef __GNUC__
+#pragma GCC diagnostic pop
+#endif
+
