diff -Naur source_newlib_original/newlib/libc/misc/fini.c source_newlib/newlib/libc/misc/fini.c
--- source_newlib_original/newlib/libc/misc/fini.c	2016-11-22 09:55:58.390369175 -0700
+++ source_newlib/newlib/libc/misc/fini.c	2016-11-22 11:12:54.119344463 -0700
@@ -13,7 +13,7 @@
 /* Handle ELF .{pre_init,init,fini}_array sections.  */
 #include <sys/types.h>
 
-#ifdef HAVE_INITFINI_ARRAY
+#if 0
 extern void (*__fini_array_start []) (void) __attribute__((weak));
 extern void (*__fini_array_end []) (void) __attribute__((weak));
 
diff -Naur source_newlib_original/newlib/libc/misc/init.c source_newlib/newlib/libc/misc/init.c
--- source_newlib_original/newlib/libc/misc/init.c	2016-11-22 09:55:58.390369175 -0700
+++ source_newlib/newlib/libc/misc/init.c	2016-11-22 11:12:54.119344463 -0700
@@ -13,7 +13,7 @@
 /* Handle ELF .{pre_init,init,fini}_array sections.  */
 #include <sys/types.h>
 
-#ifdef HAVE_INITFINI_ARRAY
+#if 0 
 
 /* These magic symbols are provided by the linker.  */
 extern void (*__preinit_array_start []) (void) __attribute__((weak));
diff -Naur source_newlib_original/newlib/libc/stdlib/calloc.c source_newlib/newlib/libc/stdlib/calloc.c
--- source_newlib_original/newlib/libc/stdlib/calloc.c	2016-11-22 09:55:58.322366817 -0700
+++ source_newlib/newlib/libc/stdlib/calloc.c	2016-11-22 11:12:54.051345345 -0700
@@ -1,6 +1,4 @@
-#ifdef MALLOC_PROVIDED
-int _dummy_calloc = 1;
-#else
+#if 0
 /*
 FUNCTION
 <<calloc>>---allocate space for arrays
diff -Naur source_newlib_original/newlib/libc/stdlib/mallocr.c source_newlib/newlib/libc/stdlib/mallocr.c
--- source_newlib_original/newlib/libc/stdlib/mallocr.c	2016-11-22 09:55:58.322366817 -0700
+++ source_newlib/newlib/libc/stdlib/mallocr.c	2016-11-22 11:12:54.047345397 -0700
@@ -1,6 +1,4 @@
-#ifdef MALLOC_PROVIDED
-int _dummy_mallocr = 1;
-#else
+#if 0
 /* ---------- To make a malloc.h, start cutting here ------------ */
 
 /* 
diff -Naur source_newlib_original/newlib/libc/stdlib/realloc.c source_newlib/newlib/libc/stdlib/realloc.c
--- source_newlib_original/newlib/libc/stdlib/realloc.c	2016-11-22 09:55:58.322366817 -0700
+++ source_newlib/newlib/libc/stdlib/realloc.c	2016-11-22 11:12:54.051345345 -0700
@@ -1,6 +1,4 @@
-#ifdef MALLOC_PROVIDED
-int _dummy_calloc = 1;
-#else
+#if 0
 /* realloc.c -- a wrapper for realloc_r.  */
 
 #include <_ansi.h>
