diff -Naur source_newlib_original/newlib/libc/string/memcpy.c source_newlib/newlib/libc/string/memcpy.c
--- source_newlib_original/newlib/libc/string/memcpy.c	2016-11-22 09:55:58.366368343 -0700
+++ source_newlib/newlib/libc/string/memcpy.c	2016-11-22 11:15:41.077635927 -0700
@@ -1,3 +1,4 @@
+#if 0
 /*
 FUNCTION
         <<memcpy>>---copy memory regions
@@ -109,3 +110,4 @@
   return dst0;
 #endif /* not PREFER_SIZE_OVER_SPEED */
 }
+#endif
diff -Naur source_newlib_original/newlib/libc/string/memset.c source_newlib/newlib/libc/string/memset.c
--- source_newlib_original/newlib/libc/string/memset.c	2016-11-22 09:55:58.366368343 -0700
+++ source_newlib/newlib/libc/string/memset.c	2016-11-22 11:15:41.077635927 -0700
@@ -1,3 +1,4 @@
+#if 0
 /*
 FUNCTION
 	<<memset>>---set an area of memory
@@ -102,3 +103,4 @@
 
   return m;
 }
+#endif
